explainability-service/src/main/java/org/kie/trustyai/service/prometheus/PrometheusScheduler.java

```diff
@@ -60,35 +68,25 @@ void calculate() {
             try {
                 for (final String modelId : getModelIds()) {
 
-                    final Predicate<Map.Entry<UUID, ReconciledMetricRequest>> filterByModelId = request -> request.getValue().getModelId().equals(modelId);
-
-                    final List<Map.Entry<UUID, ReconciledMetricRequest>> modelSpdRequest =
-                            spdRequests.entrySet().stream().filter(filterByModelId).collect(Collectors.toList());
+                    final Predicate<Map.Entry<UUID, BaseMetricRequest>> filterByModelId = request -> request.getValue().getModelId().equals(modelId);
 
-                    final List<Map.Entry<UUID, ReconciledMetricRequest>> modelDirRequest =
-                            dirRequests.entrySet().stream().filter(filterByModelId).collect(Collectors.toList());
+                    List<Map.Entry<UUID, BaseMetricRequest>> requestsSet = getAllRequestsFlat().entrySet().stream().filter(filterByModelId).collect(Collectors.toList());
 
                     // Determine maximum batch requested. All other batches as sub-batches of this one.
-                    final int maxBatchSize = Stream.concat(modelSpdRequest.stream(), modelDirRequest.stream())
+                    final int maxBatchSize = requestsSet.stream()
                             .mapToInt(entry -> entry.getValue().getBatchSize()).max()
                             .orElse(serviceConfig.batchSize().getAsInt());
 
                     final Dataframe df = dataSource.get().getDataframe(modelId, maxBatchSize);
 
-                    // SPD requests
-                    modelSpdRequest.forEach(entry -> {
+                    requestsSet.forEach(entry -> {
+                        // entry value: BaseMetricRequest
                         final Dataframe batch = df.tail(Math.min(df.getRowDimension(), entry.getValue().getBatchSize()));
-                        final double spd = calculator.calculateSPD(batch, entry.getValue());
-                        publisher.gaugeSPD(entry.getValue(), modelId, entry.getKey(), spd);
-                    });
+                        String metricName = entry.getValue().getMetricName();
+                        final double value = metricsDirectory.getCalculator(metricName).apply(batch, entry.getValue());
 
-                    // DIR requests
-                    modelDirRequest.forEach(entry -> {
-                        final Dataframe batch = df.tail(Math.min(df.getRowDimension(), entry.getValue().getBatchSize()));
-                        final double dir = calculator.calculateDIR(batch, entry.getValue());
-                        publisher.gaugeDIR(entry.getValue(), modelId, entry.getKey(), dir);
+                        publisher.gauge(entry.getValue(), modelId, entry.getKey(), value);
```

Comment:
I don't know if it is an issue but this refactor makes the assumption that each metric is gauge.
I think it is safe as assumption, I cannot at the moment think at a fairness metrics that can be represented with histogram or counter ( https://prometheus.io/docs/concepts/metric_types/ ) 🤔

Reply:
Opened a ticket https://github.com/trustyai-explainability/trustyai-explainability/issues/299